test/test-node.js

```diff
@@ -27,8 +115,89 @@ describe('rclnodejs node test suite', function() {
     rclnodejs.shutdown();
   });
 
-  it('Try creating a node', function() {
-    const node = rclnodejs.createNode('example_node');
-    rclnodejs.spin(node);
+  beforeEach(function() {
+    node = rclnodejs.createNode('my_node', '/my_ns');
+    rclString = rclnodejs.require('std_msgs').msg.String;
+  });
+
+  afterEach(function() {
+    node.destory();
+  });
+
+  it('node setter/getter', function() {
+    assert.notDeepStrictEqual(null, node);
+
+    // assert.deepStrictEqual(node.get_name(), 'my_node');
+    // assert.deepStrictEqual(node.get_namespace(), '/my_ns');
+  });
+
+  it('node.createPublisher', function() {
+    node.createPublisher(rclString, 'chatter');
+
+    assert.throws(function() {
+      node.createPublisher(rclString, 'chatter?');
+    }, /topic name is invalid/, 'Failed to createPublisher');
+
+    assert.throws(function() {
+      node.createPublisher(rclString, '/chatter/42_is_the_answer');
+    }, /topic name is invalid/, 'Failed to createPublisher');
+
+    assert.throws(function() {
+      node.createPublisher(rclString, '/chatter/{bad_sub}');
+    }, /unknown substitution/, 'Failed to createPublisher');
+  });
+
+  it('node.createSubscription', function() {
+    node.createSubscription(rclString, 'chatter', () => {
+      assert.ok(true);
+    });
+
+    assert.throws(function() {
+      node.createSubscription(rclString, 'chatter?', () =>{});
+    }, /topic name is invalid/, 'Failed to createSubscription');
+
+    assert.throws(function() {
+      node.createSubscription(rclString, '/chatter/42_is_the_answer', () =>{});
+    }, /topic name is invalid/, 'Failed to createSubscription');
+
+    assert.throws(function() {
+      node.createSubscription(rclString, '/chatter/{bad_sub}', () =>{});
+    }, /unknown substitution/, 'Failed to createSubscription');
+  });
+
+  it('node.createClient', function() {
+    var GetParameters = rclnodejs.require('rcl_interfaces').srv.GetParameters;
+    node.createClient(GetParameters, 'get/parameters');
+
+    assert.throws(function() {
+      node.createClient(GetParameters, 'get/parameters?');
+    }, /topic name is invalid/, 'Failed to createClient');
+
+    assert.throws(function() {
+      node.createClient(GetParameters, 'get/42parameters');
+    }, /topic name is invalid/, 'Failed to createClient');
+
+    assert.throws(function() {
+      node.createClient(GetParameters, 'foo/{bad_sub}');
+    }, /unknown substitution/, 'Failed to createClient');
+  });
+
+  it('node.createService', function() {
+    var GetParameters = rclnodejs.require('rcl_interfaces').srv.GetParameters;
+    node.createService(GetParameters, 'get/parameters', () => {
+      assert.ok(true);
+    });
+
+    assert.throws(function() {
```

Comment:
Move invalid names to an array and use `foreach` to invoke assertion to make test robust.